File: backend/Multibanca.Domain.Models/Multibanca/BBVA/Escrituracion/realizar_vb_final_abogado.cs
namespace Multibanca.Domain.Models.Multibanca.BBVA.Escrituracion;

public class realizar_vb_final_abogado
{
    public long     id { get; set; }
    public long     id_expediente { get; set; }
    public string?  id_actividad { get; set; }

    public string?  requiere_devolucion { get; set; }
    public string?  tipologia { get; set; }
    public string?  casuistica { get; set; }
    public string?  origen_tramite { get; set; }
    public bool     bandera_excepcion { get; set; }
    public string?  observaciones { get; set; }

    public bool      is_active { get; set; }
    public bool      row_status { get; set; }
    public int       created_by { get; set; }
    public DateTime  created_date { get; set; }
    public int?      modified_by { get; set; }
    public DateTime? modified_date { get; set; }
}

File: frontend/src/features/actividades/realizar_vb_final_abogado/models/realizar_vb_final_abogado.ts
import type { Auditoria } from '@/models/Auditoria';

export interface RealizarVBFinalAbogado extends Auditoria {
  id: number;
  id_expediente: number;
  id_actividad: string;
  requiere_devolucion: string | null;
  tipologia: string | null;
  casuistica: string | null;
  origen_tramite: string | null;
  bandera_excepcion: boolean;
  observaciones: string | null;
}

export interface GetByExpedienteResponse {
  formulario: RealizarVBFinalAbogado;
  origen_tramite: string;
  bandera_excepcion: boolean;
}

export const EMPTY_REALIZAR_VB_FINAL_ABOGADO = (id_expediente: number): RealizarVBFinalAbogado => ({
  id: 0, id_expediente, id_actividad: 'BBVA_ESCRITURACION_REALIZAR_VB_FINAL_ABOGADO',
  requiere_devolucion: 'NO', tipologia: null, casuistica: null,
  origen_tramite: null, bandera_excepcion: false, observaciones: null,
  is_active: true, row_status: true, created_by: 0, created_date: '', modified_by: null, modified_date: null,
});
